pkc

```diff
@@ -54,26 +54,26 @@ check_wallet_address() {
 
 check_dependencies() {
     # Check Docker installation
-    if [[ -x "$(command -v docker)" ]]; then
-        true
-    else
-        echo "Docker is not yet installed. Aborting PKC installation."
+    if [[ ! -x "$(command -v docker)" ]]; then
+        echo "Error: Docker is not yet installed. Aborting PKC installation." >&2
         exit 1
     fi
 
     # Check docker-compose installation
-    if [[ -x "$(command -v docker-compose)" ]]; then
-        true
-    else
-        echo "docker-compose is not yet installed. Aborting PKC installation."
+    if [[ ! -x "$(command -v docker-compose)" ]]; then
+        echo "Error: docker-compose is not yet installed. Aborting PKC installation." >&2
         exit 1
     fi
 
     # Check git installation
-    if [[ -x "$(command -v git)" ]]; then
-        true
-    else
-        echo "Git is not yet installed. Aborting PKC installation."
+    if [[ ! -x "$(command -v git)" ]]; then
+        echo "Error: Git is not yet installed. Aborting PKC installation." >&2
+        exit 1
+    fi
+
+    # Check Docker Engine running
+    if [[ $(docker version) != *"Engine"* ]]; then
```

Comment:
I tested this by doing:
- `sudo systemctl stop docker` also `sudo systemctl disable docker`
- `docker version`
It outputs the text `Engine` in both the `stop` and `disable` cases.
It seems this check is macOS-specific? Also, shouldn't the goal be detecting whether the Docker daemon is both enabled and running?

Reply:
thanks for testing on linux. sounds like it is indeed environment specific. will look for an alternative so we don't have to fork code based on platform